File: GC.Domain/Contacts/ForeignContacts/ForeignContact.cs
using GC.Domain.Persons;
using System;

namespace GC.Domain.Contacts.ForeignContacts
{
    public class ForeignContact : FromPerson
    {
        public Guid Id { get; }
        public ForeignContactType Type { get; }
        public String FirstName { get; }
        public String? MiddleName { get; }
        public String LastName { get; }
        public String PhoneNumber { get; }

        public ForeignContact(Guid id, ForeignContactType type, String firstName, String? middleName, String lastName, String phoneNumber)
        {
            Id = id;
            Type = type;
            FirstName = firstName;
            MiddleName = middleName;
            LastName = lastName;
            PhoneNumber = phoneNumber;
        }
    }
}

File: GC.WebSpace/ClientApp/domain/contacts/foreignContacts/foreignContact.ts
import { ForeignContactType } from "./foreignContactType";

export class ForeignContact {
    constructor(
        public readonly id: string,
        public readonly type: ForeignContactType,
        public readonly firstName: string,
        public readonly middleName: string,
        public readonly lastName: string,
        public readonly phoneNumber: string
    ) { }

    get fullName() {
        return `${this.lastName} ${this.firstName} ${this.middleName}`;
    }
}

export namespace ForeignContact {
    export function toModel(object: any): ForeignContact {
        return new ForeignContact(object.id, object.type, object.firstName, object.middleName, object.lastName, object.phoneNumber);
    }

    export function toModels(objects: any[]): ForeignContact[] {
        return objects.map(toModel);
    }
}